internal/util/pkgutil/pkgutil_test.go

```diff
@@ -105,6 +106,181 @@ func TestWalkPackage(t *testing.T) {
 	}
 }
 
+func TestCopyPackage(t *testing.T) {
+	testCases := map[string]struct {
+		pkg               *pkgbuilder.RootPkg
+		copyRootKptfile   bool
+		subpackageMatcher pkg.SubpackageMatcher
+		expected          []string
+	}{
+		"subpackages without root kptfile": {
```

Comment:
Add some tests here around deeply nested packages.

Reply:
Added positive cases surrounding nested packages and their types. Let me know if more are desired.